examples/app_boring/app.py

```diff
@@ -43,6 +43,10 @@ def __init__(self):
             raise_exception=True,
         )
 
+    @property
+    def ready(self) -> bool:
+        return self.source_work.has_succeeded and self.dest_work.is_running
```

Comment:
The second one is conditioned on the first one to succeeded already in the flow.

```suggestion
        return self.dest_work.is_running
```